content/en/docs/concepts/cloud-provider/ccm-webhook.md

```diff
@@ -0,0 +1,140 @@
+content_type: concept
+weight: 20
+---
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.27" state="alpha" >}}
+
+The Kubernetes {{< glossary_tooltip text="cloud provider"
+term-id="cloud-provider" >}} webhook framework is designed to make it easier
+for cloud providers to serve {{< glossary_tooltip text="admission webhooks"
+term-id="admission-webhook" >}} from their cloud controller manager binaries.
+
+<!-- body -->
+
+## Overview
+
+In order to take advantage of the cloud controller manager webhook framework, a
+vendor needs to use the new `app.CommandBuilder` imported from
+`k8s.io/cloud-provider/app` to generate their cloud controller manager's
+`cobra.Command`.  This command is used to start thloud controller manager
+binary, and currently most ccms use `NewCloudControllerManagerCommand()` to
+generate the `cobra.Command`.  The new alternative, found in
+[builder.go](https://github.com/kubernetes/kubernetes/blob/d89d5ab2680bc74fe4487ad71e514f4e0812d9ce/staging/src/k8s.io/cloud-provider/app/builder.go)
+supports the new webhook functionality in a way that no longer requires cloud
+providers to update their cloud controller manager binary when new features are
+released.
+
+## Example Usage
+
+In the main package of your cloud controller manager binary (or wherever you
+construct your `cobra.Command` using `NewCloudControllerManagerCommand()`), you
+must first migrate your command creation process to use the `CommandBuilder`.
+
+With the `CommandBuilder` pattern in place, the `RegisterWebhook()` function can be called to complete the registration of the webhook.
+An example `main()` function might look like this:
+
+```go
+func main() {
+        rand.Seed(time.Now().UTC().UnixNano())
+
+        logs.InitLogs()
+        defer logs.FlushLogs()
+
+        fss := cliflag.NamedFlagSets{}
+
+        examplecontroller.Options.AddFlags(fss.FlagSet("example controller"))
+
+        builder := app.NewBuilder()
+        builder.AddFlags(fss)
+        builder.RegisterDefaultControllers()
+        builder.RegisterController(examplecontroller.ExampleControllerKey, examplecontroller.ExampleControllerConstructor)
+        builder.RegisterWebhook("example-webhook", app.WebhookConfig{
+                Path:             examplewebhook.Path,
+                AdmissionHandler: examplewebhook.ValidatingWebhook,
+        })
+        builder.SetStopChannel(wait.NeverStop)
+        builder.SetCloudInitializer(cloudInitializer)
+        builder.SetProviderDefaults(cpoptions.ProviderDefaults{
+                WebhookBindPort: &examplewebhook.Webhookport,
+        })
+        command := builder.BuildCommand()
+
+        if err := command.Execute(); err != nil {
+                klog.Fatalf("unable to execute command: %v", err)
+        }
+}
+```
+
+Finally, a webhook configuration needs to be created, referencing the webhook endpoint which was just created.
+
+```yaml
+// Todo a webhook configuration yaml
```

Comment:
explain this TODO a bit more please

Reply:
That should be explained by the previous sentence, so I deleted the comment.